x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_page_header.tsx

```diff
@@ -140,9 +142,31 @@ const useFeatureFlagTabs = () => {
   };
 };
 
+const useLogsOnlyTabs = () => {
+  const { asset } = useAssetDetailsRenderPropsContext();
+  const { dataStreams } = useEntitySummary({
+    entityType: asset.type,
+    entityId: asset.id,
+  });
+
+  const isMetrics = isMetricsSignal(dataStreams);
+
+  const hasMetricsTab = useCallback(
+    (tabItem: Tab) => {
+      return isMetrics || tabItem.id !== ContentTabIds.METRICS;
+    },
+    [isMetrics]
+  );
+
+  return {
+    hasMetricsTab,
+  };
+};
+
 const useTabs = (tabs: Tab[]) => {
   const { showTab, activeTabId } = useTabSwitcherContext();
   const { isTabEnabled } = useFeatureFlagTabs();
+  const { hasMetricsTab } = useLogsOnlyTabs();
```

Comment:
Can you rename this hook? I think, naming it `useLogsOnlyTabs` and returning `hasMetricsTab` feels wrong.

Reply:
Oops! Good catch